tests/test_algorithm.py

```diff
@@ -2018,3 +2022,99 @@ def test_remove_data(self):
         # initially only data for X should be sent and on the last day only
         # data for Y should be sent since X is expired
         np.testing.assert_array_equal(self.algo.data, expected_lengths)
+
+
+class TestAssetAutoClose(TestCase):
+    """
+    Tests if delisted securities are properly removed from a portfolio holding
+    positions in said securities.
+    """
+    def setUp(self):
+        dt = pd.Timestamp('2015-01-05', tz='UTC')
+        self.env = TradingEnvironment()
+        ix = self.env.trading_days.get_loc(dt)
+
+        self.metadata = make_rotating_equity_info(
+            3, dt, trading_day, 2, 2,
+        )
+
+        # Make all start dates equal.
+        self.metadata['start_date'] = [self.metadata['start_date'][0]]*3
+
+        # Add an auto close date column, where liquidation occurs the day after
+        # the end date.
+        self.metadata['auto_close_date'] = \
+            self.metadata['end_date'] + trading_day
+
+        index_x = self.env.trading_days[ix:ix + 3]
+        data_x = pd.DataFrame(
+            data=[[1, 100], [2, 100], [3, 100]],
+            index=index_x,
+            columns=['price', 'volume'],
+        )
+        index_y = self.env.trading_days[ix:ix + 5]
+        data_y = pd.DataFrame(
+            data=[[4, 100], [5, 100], [6, 100], [7, 100], [8, 100]],
+            index=index_y,
+            columns=['price', 'volume'],
+        )
+        index_z = self.env.trading_days[ix:ix + 7]
+        data_z = pd.DataFrame(
+            data=[
+                [9, 100], [10, 100], [11, 100], [12, 100], [13, 100],
+                [14, 100], [15, 100],
+            ],
+            index=index_z,
+            columns=['price', 'volume'],
+        )
+
+        pan = pd.Panel({0: data_x, 1: data_y, 2: data_z})
+        self.source = DataPanelSource(pan)
+
+    def tearDown(self):
+        del self.env
+
+    def test_delisted_securities(self):
+        """
+        Make sure that after a security gets delisted, our portfolio holds the
+        correct number of securities and correct amount of cash.
+        """
+        self.env.write_data(equities_df=self.metadata)
```

Comment:
You're calling `write_data` in all of these tests with `self.metadata`.  Can we move that logic into the shared setup (it should probably be in a `setUpClass` for performance)?